UI/FleetWnd.cpp

```diff
@@ -2769,10 +2771,33 @@ int FleetDetailPanel::ShipInRow(GG::ListBox::iterator it) const {
 ////////////////////////////////////////////////
 // FleetWnd
 ////////////////////////////////////////////////
-FleetWnd::FleetWnd(const std::vector<int>& fleet_ids, bool order_issuing_enabled,
-         int selected_fleet_id/* = INVALID_OBJECT_ID*/,
-         GG::Flags<GG::WndFlag> flags/* = INTERACTIVE | DRAGABLE | ONTOP | CLOSABLE | RESIZABLE*/,
-         const std::string& config_name) :
+namespace {
+    /** \p create or grow a bounding \p box from \p pt. */
+    GG::Rect CreateOrGrowBox(bool create, const GG::Rect box, const GG::Pt pt) {
+        if (create)
+            return GG::Rect(pt, pt);
+        else
+            return GG::Rect(
+                std::min(box.Left(),    pt.x),
+                std::min(box.Top(),     pt.y),
+                std::max(box.Right(),   pt.x),
+                std::max(box.Bottom(),  pt.y));
+    }
+
+    /** Can \p bounds encompass \p candidate? */
+    bool CouldContain(GG::Rect bounds, GG::Rect candidate) {
```

Comment:
somewhat misleading function name... doesn't check if `candidate` is or could be within, just that it is smaller or equal in size.

Reply:
If I am understanding right, this is being used to check if fleets sharing a FleetWnd have dispersed too far, from one turn to the next, to still share the FleetWnd, and that suggests a names to me like ```FleetsHaveNotDispersed``` or ```FleetsStillClustered```.

Reply:
The function itself has nothing to do with fleets, so I'd have its name describe what it actually does, not how it's used in one case.

Reply:
Name suggestions seem easier to come up with if args are swapped: `CouldRelocateWithin`

Reply:
> so I'd have its name describe what it actually does, not how it's used in one case.

It seems to me that I am proposing to have the name describe what it actually does in the most significant sense-- the determination that it is making, rather than the details of the logic used to make that determination.  And I am puzzled about the "not how it's used in one case" -- it appears to me that this function is used in exactly two places, and in both places it is used the same general way with the same the significance.

The name ```CouldRelocateWithin``` could be technically accurate as you suggest, but it would seem to me to imply a different significance and purpose for the function than what it actually has, which would make code harder to understand rather than easier.

Reply:
In this sense, I'd expect `FleetsX(non-Fleet args)` to derive fleets from args.
Would view this function with a potential to move into GG/PtRect as non-/member function if later desired for other cases.

Reply:
I'd prefer `SmallerThan` and `BiggerThan` for this case.

Reply:
I changed it to `bool SmallerOrEqual(GG::Rect ll, GG::Rect rr)`.